modin/pandas/groupby.py

```diff
@@ -295,7 +295,7 @@ def __getitem__(self, key):
                 and all(c in self._columns for c in self._by.columns)
                 and self._drop
             ):
-                key = [key] + list(self._by.columns)
+                key = list(self._by.columns) + [key]
```

Comment:
Why is this change needed? It should not impact anything from what I see.

Reply:
this is the order in which pandas places columns at `groupby.__getitem__`, look at this reproducer:
```python
if __name__ == "__main__":
    import modin.pandas as pd
    import pandas
    import numpy as np

    data = {
        "col1": [0, 1, 2, 3],
        "col2": [4, 5, np.NaN, 7],
        "col3": [np.NaN, np.NaN, 12, 10],
        "col4": [17, 13, 16, 15],
        "col5": [-4, -5, -6, -7],
    }

    pd_grp = pandas.DataFrame(data).groupby("col4", as_index=False)
    md_grp = pd.DataFrame(data).groupby("col4", as_index=False)

    print("pd_columns:", pd_grp["col1"].count().columns)   # pd_columns: Index(['col4', 'col1'], dtype='object')
    print("md_columns:", md_grp["col1"].count().columns)  # md_columns: Index(['col1', 'col4'], dtype='object') 
```

Reply:
Yes, I see.